core/test/qa_Block.cpp

```diff
@@ -16,6 +15,166 @@ template<>
+
+    T
+    processOne(T) const {
+        return T();
+    }
+};
+
+ENABLE_REFLECTION_FOR_TEMPLATE(BlockSignaturesProcessOneConst, in, out);
+static_assert(gr::HasRequiredProcessFunction<BlockSignaturesProcessOneConst<float>>);
+static_assert(gr::HasProcessOneFunction<BlockSignaturesProcessOneConst<float>>);
+static_assert(gr::HasConstProcessOneFunction<BlockSignaturesProcessOneConst<float>>);
+static_assert(!gr::HasProcessBulkFunction<BlockSignaturesProcessOneConst<float>>);
+
+enum class ProcessBulkVariant { VARIANT_SPAN, VARIANT_PUBLISHABLE_SPAN, VARIANT_PUBLISHABLE_SPAN2, VARIANT_CONSUMABLE_SPAN, VARIANT_CONSUMABLE_SPAN2 };
+
+template<typename T, ProcessBulkVariant processVariant>
+struct BlockSignaturesProcessBulkSpan : public gr::Block<BlockSignaturesProcessBulkSpan<T, processVariant>> {
+    gr::PortIn<T>  in{};
+    gr::PortOut<T> out{};
+
+    gr::work::Status
+    processBulk(std::span<const T>, std::span<T>)
+        requires(processVariant == ProcessBulkVariant::VARIANT_SPAN)
+    {
+        // do some bulk-type processing
+        return gr::work::Status::OK;
+    }
+
+    gr::work::Status
+    processBulk(std::span<const T>, gr::PublishableSpan auto &)
+        requires(processVariant == ProcessBulkVariant::VARIANT_PUBLISHABLE_SPAN)
+    {
+        // do some bulk-type processing
+        return gr::work::Status::OK;
+    }
+
+    gr::work::Status
+    processBulk(std::span<const T>, gr::PublishableSpan auto)
+        requires(processVariant == ProcessBulkVariant::VARIANT_PUBLISHABLE_SPAN2)
+    {
+        // do some bulk-type processing
+        return gr::work::Status::OK;
+    }
+
+    gr::work::Status
+    processBulk(gr::ConsumableSpan auto, std::span<T>)
+        requires(processVariant == ProcessBulkVariant::VARIANT_CONSUMABLE_SPAN)
+    {
+        // do some bulk-type processing
+        return gr::work::Status::OK;
+    }
+
+    gr::work::Status
+    processBulk(gr::ConsumableSpan auto &, std::span<T>)
+        requires(processVariant == ProcessBulkVariant::VARIANT_CONSUMABLE_SPAN2)
+    {
+        // do some bulk-type processing
+        return gr::work::Status::OK;
+    }
+};
+
+ENABLE_REFLECTION_FOR_TEMPLATE_FULL((typename T, ProcessBulkVariant processVariant), (BlockSignaturesProcessBulkSpan<T, processVariant>), in, out);
+using enum ProcessBulkVariant;
+static_assert(gr::HasRequiredProcessFunction<BlockSignaturesProcessBulkSpan<float, VARIANT_SPAN>>);
+static_assert(!gr::HasProcessOneFunction<BlockSignaturesProcessBulkSpan<float, VARIANT_SPAN>>);
+static_assert(!gr::HasConstProcessOneFunction<BlockSignaturesProcessBulkSpan<float, VARIANT_SPAN>>);
+static_assert(gr::HasProcessBulkFunction<BlockSignaturesProcessBulkSpan<float, VARIANT_SPAN>>);
+static_assert(gr::HasProcessBulkFunction<BlockSignaturesProcessBulkSpan<float, VARIANT_PUBLISHABLE_SPAN>>);
+static_assert(!gr::HasProcessBulkFunction<BlockSignaturesProcessBulkSpan<float, VARIANT_PUBLISHABLE_SPAN2>>); // TODO: fix the signature is required to work
```

Comment:
I really like this unit test 👍  Looking ahead, it would be beneficial to include test cases for arrays of input/output ports as well. This addition would not only enhance our test coverage but also serve as a valuable example for users to understand what signature are possible.